### PR TITLE
Issue 22287 writing osgi extras

### LINKED ISSUE #22287: Reading osgi import packages fails when osgi version specifies a range
When an osgi plugin/fragment is uploaded, we attempt to read the packages and automatically add them to our `osgi-extras` file.  The issue is that we are doing a simple split of the string on a comma.  OSGi allows the version of packages to specify a range, and this range is in quotes and has a comma in it.  The import looks like:

```
...
 nguagesmanager.model,com.dotmarketing.util,com.liferay.portal.model,c
 om.liferay.portal.util,io.vavr;version="[0.10,1)",javax.servlet.http;
 version="[3.1,4)",org.apache.velocity,org.apache.velocity.app,org.apa
 che.velocity.context,org.apache.velocity.exception,org.apache.velocit
 y.runtime,org.apache.velocity.runtime.directive,org.apache.velocity.r
 untime.parser.node,org.osgi.framework;version="[1.8,2)"
```

Uploading a plugin that has imports like the above blows osgi up.

### PR DESCRIPTION
**Pull request by wezell**
This pull request prevents writing a version range to the osgi exported packages.

### CODE REVIEW

File: dotCMS/src/main/java/com/dotmarketing/portlets/osgi/AJAX/OSGIAJAX.java [hunk 1]
```diff
@@ -163,6 +164,9 @@ public void add ( HttpServletRequest request, HttpServletResponse response ) thr
                     IOUtils.closeQuietly( in );
                 }
             }
+            OSGIUtil.getInstance().checkUploadFolder();
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/osgi/AJAX/OSGIAJAX.java:167]: Very good

File: dotCMS/src/main/java/com/dotmarketing/portlets/osgi/AJAX/OSGIAJAX.java [hunk 2]
```diff
@@ -201,12 +205,10 @@ public void modifyExtraPackages ( HttpServletRequest request, HttpServletRespons
...
         String extraPackages = request.getParameter( "packages" );
 
-        //Override the file with the values we just read
-        BufferedWriter writer = new BufferedWriter( new FileWriter( OSGIUtil.getInstance().FELIX_EXTRA_PACKAGES_FILE ) );
-        writer.write( extraPackages );
-        writer.close();
-        Logger.info( OSGIAJAX.class, "OSGI Extra Packages Saved");
-        //Send a response
+
+        OSGIUtil.getInstance().writeOsgiExtras(extraPackages);
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/osgi/AJAX/OSGIAJAX.java:209]: Nice

**Review by jdotcms — APPROVED** [review]
Great Job

**Comment by jdotcms** [comment]
I have reviewed the functionality
The upload button is working
the write Osgi extra too
and did not find issues on the imported packages

**Review by dsilvam — APPROVED** [review]

File: dotCMS/src/main/java/com/dotmarketing/util/StringUtils.java [hunk 3]
```diff
@@ -65,6 +65,42 @@ public static boolean isHtml(final String htmlString) {
     }
 
 
+    public static List<String> splitOnCommasWithQuotes(String importPackage){
```
Comment (nollymar) [inline dotCMS/src/main/java/com/dotmarketing/util/StringUtils.java:68]: This method should have a test

**Pull request merged**